docs/persistence.md

```diff
@@ -0,0 +1,147 @@
+---
+description: __noun__ firm or obstinate continuance in a course of action in spite of difficulty or opposition.
+---
+
+# Persistence
+
+Planning the design of StimulusReflex user interfaces is simpler than what we're used to, whether that's submitting a form and loading the results or using a JavaScript SPA framework to communicate with an API. Since your applications won't have a client state, we estimate that 80% of the effort previously required is now behind you. It does, however, require some unlearning and the willingness to rethink how you approach persisting the state of your application. We recognize that this can be jarring at first and that even positive changes can feel like work.
+
+The best way to start thinking about how interfaces are designed in a StimulusReflex world is to remember that when you access a URL in your browser, you are seeing the current state of your UI; there is no mounting process. Your request goes through the Rails router to Action Pack where your controller renders your view templates and sends them down the wire. At this stage, StimulusReflex is not yet involved. You can call ActiveRecord, access data from Redis or the Rails cache, and set instance variables that get picked up in your view. This is Rails in full 2004 glory.
+
+Once the page is displayed in your browser, the magic begins. StimulusReflex opens an ActionCable (websockets) connection back to the server and keeps it open, waiting for events. It then scans your document for `data-reflex` attributes so that it can wire up those events to Stimulus controllers that are responsible for mapping events in your browser to methods in your Reflex classes on the server.
+
+When Reflex method is called, it is able to access ActiveRecord, Redis, the session object and Rails cache. After the Reflex method is complete the Rails controller's action method is called and any instance variables set are passed on to the Rails controller's action method. In this way, it's similar to a `before_action` filter. However, this is also where a lot of people new to StimulusReflex get confused. The order of operations can seem fuzzy.
+
```

Comment:
`s/this is also where a lot of people ... get confused/this is also where people ... can get confused`

Reply:
Just to be clear... do you actually want ellipses?

Reply:
No. Was just being terse.